File: LeetCode/566. Reshape the Matrix.cs
public class Solution {
    public int[][] MatrixReshape(int[][] mat, int r, int c) {
        
        int rows=mat.Length;
        int columns=mat[0].Length;
        int size=columns*rows;
        int[][] array = new int[r][];

      
            
        if(rows==r && columns==c){return mat;}
        if(r*c==size){
            int i=0;
            int j=0;
            int count=0;
            for(int a=0;a<r;a++){
                
                array[a] = new int[c];
                
                for(int b=0;b<c;b++){

                    if(count==columns){i++;j=0;count=0;}
                    
                    array[a][b]=mat[i][j];
                    count++;j++;
            }}
            return array;
        }
        return mat;
    }
}

File: Data Structures and Algorithms/LeetCode/566. Reshape the Matrix.py
class Solution:
    def matrixReshape(self, mat: List[List[int]], r: int, c: int) -> List[List[int]]:
        size = len(mat)*len(mat[0])
        matrix = [[0 for j in range(c)] for i in range(r)]
        if size != r*c:
            return mat
        countc = 0
        countr = 0
        for i in mat:
            for j in i:
                if countc == c:
                    countc = 0
                    countr += 1
                matrix[countr][countc] = j
                countc += 1
        return matrix
